### ISSUE TITLE
Prevent toggling from visual to source from changing source for Rmd and qmd

### ISSUE THREAD

**Issue by jtlandis** [rstudio/rstudio#14126]
I've been using RStudio since I starting learning R, and I'm happy to see all of the new features being added like the visual editor. I am wondering though if there is a way to turn off RStudio automatically rewriting your document as you toggle between views of a visual editor and source.

I was in the process of writing up a document in source and I have been writing each sentence on its own line. I decided to write like this as it is easier to visualize what has changed between commits than looking at an entire paragraph (on one line) when all that changed was punctuation. Furthermore, putting each sentence on a new line in source view had no affect on how the paragraph was rendered, all the sentences are still concatenated together. 👍 

I was about to implement a table, so I swapped to the visual editor because it is much easier (and quicker) to use. When I toggled back, I noticed that all my sentences that were on their on lines had been moved to a single line. 👎 

In short, is there a way to turn this rewriting between views off? This has happened enough to where 

This isn't the only example I have either. Swapping between labeling with foot notes like [^longnote], will reindex all of the footnotes numerically to [^1] or whatever the next appropriate index would be.

**Comment by ronblum** [comment]
@jtlandis Thank you for raising this! The visual editor uses [Pandoc standard formatting](https://rstudio.github.io/visual-markdown-editing/markdown.html). Does setting the document [to use canonical mode](https://quarto.org/docs/visual-editor/markdown.html#canonical-mode) address what you're looking for?

**Comment by jtlandis** [comment]
This is exactly what I was looking for! I wish I had known about this before. Oh well, it will be good to remember for my next quarto project.
